### PR TITLE
UIREC-190 Validate user has made a change to the piece form before ac…

### PR DESCRIPTION
**Pull request by usavkov-epam**
…tivating 'Save' or 'Quick receive' buttons

## Purpose

https://issues.folio.org/browse/UIREC-190
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/TitleDetails/AddPieceModal/AddPieceModal.js [hunk 1]
```diff
@@ -396,6 +400,8 @@ AddPieceModal.propTypes = {
   locations: PropTypes.arrayOf(PropTypes.object),
   poLine: PropTypes.object.isRequired,
   getHoldingsItemsAndPieces: PropTypes.func.isRequired,
+  initialValues: PropTypes.object.isRequired,
+  pristine: PropTypes.bool,
```
Comment (AndreiShumski2) [inline src/TitleDetails/AddPieceModal/AddPieceModal.js:404]: isRequired

File: src/TitleDetails/AddPieceModal/AddPieceModalContainer.js [hunk 2]
```diff
@@ -46,13 +46,20 @@ const AddPieceModalContainer = ({
     ? PIECE_FORMAT_OPTIONS.filter(({ value }) => [PIECE_FORMAT.electronic, PIECE_FORMAT.physical].includes(value))
     : PIECE_FORMAT_OPTIONS.filter(({ value }) => value === initialValues.format);
 
+  const initialCheckboxValues = ['discoverySuppress', 'isCreateAnother', 'isCreateItem', 'supplement']
```
Comment (AndreiShumski2) [inline src/TitleDetails/AddPieceModal/AddPieceModalContainer.js:49]: useMemo?

**Review by AndreiShumski2 — APPROVED** [review]

File: src/TitleDetails/AddPieceModal/AddPieceModal.js (hunk 1 above)
Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.js]: updated

File: src/TitleDetails/AddPieceModal/AddPieceModalContainer.js (hunk 2 above)
Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModalContainer.js]: updated

**Review by NikitaSedyx — APPROVED** [review]

File: src/TitleDetails/AddPieceModal/AddPieceModalContainer.js [hunk 3]
```diff
@@ -46,13 +46,22 @@ const AddPieceModalContainer = ({
     ? PIECE_FORMAT_OPTIONS.filter(({ value }) => [PIECE_FORMAT.electronic, PIECE_FORMAT.physical].includes(value))
     : PIECE_FORMAT_OPTIONS.filter(({ value }) => value === initialValues.format);
 
+  const initialCheckboxValues = useMemo(() => (
```
Comment (NikitaSedyx) [inline src/TitleDetails/AddPieceModal/AddPieceModalContainer.js:49]: FYI interest, useMemo here is not really required as you do `{ ...initialValues, ...initialCheckboxValues }`, with memo only only optimize object calculation and it's not big calculation

**Pull request merged**